crates/bevy_ecs/src/entity/clone_entities.rs

```diff
@@ -175,29 +185,51 @@ impl<'w> EntityCloneBuilder<'w> {
             filter_allows_components,
             filter: Arc::new(filter),
             clone_handlers_overrides: Arc::new(clone_handlers_overrides),
+            move_components,
         }
         .clone_entity(world);
 
         world.flush_commands();
     }
 
+    /// Allows for a scoped mode where any changes to the filter that allow/deny components
+    /// will also allow/deny those components' required components, recursively.
+    pub fn with_required_components(
+        &mut self,
+        builder: impl FnOnce(&mut EntityCloneBuilder) + Send + Sync + 'static,
+    ) -> &mut Self {
+        self.attach_required_components = true;
+        builder(self);
+        self.attach_required_components = false;
+        self
+    }
+
+    /// Sets the cloner to remove any components that were cloned,
+    /// effectively moving them from the source entity to the target.
+    ///
+    /// This only applies to components that are allowed through the filter
+    /// at the time [`EntityCloneBuilder::clone_entity`] is called.
+    pub fn enable_move_on_clone(&mut self) -> &mut Self {
```

Comment:
Bikeshed: I think these methods might be clearer if they were `move_components` and `keep_components`. The enable/disable terminology was confusing to me at first.

Reply:
I think this should just be one method: `move_components` that takes a bool. That way it would be in line with the other `EntityCloneBuilder` options (like `add_observers`). It was designed to look somewhat similar to `std::fs::OpenOptions`.